influx/query.js

```diff
@@ -1,55 +1,108 @@
+*
+* @param datasource - name of the datasource to check
+*
+* @returns Promise
+*/
+module.exports.checkDataSourceExists = function(datasource) {
+  return client.getDatabaseNames()
+    .then(names => {
+      return (names.indexOf(datasource) >= 0) ? true : false;
+    })
+    .catch(err => {
+      console.error('Error getting InfluxDB database names! ' + err);
+    });
+};
 
-    // or single-host configuration
-    host: 'localhost',
-    port: 8086,
-    protocol: 'http',
-    database: datasource
+/**
+* @desc Create database with datasource name.
+*
+* @param datasource - name of the datasource to check
+*
+* @returns Promise
+*/
+module.exports.createDatabase = function(datasource) {
+  return client.createDatabase(datasource).then(()  => {
+    console.log('Created database for datasource: ' + datasource);
+  })
+  .catch(err => {
+    console.log('Error trying to create database: ' + err);
   });
+};
 
-  switch (datasource) {
-    case 'github':
-      getGithubData(measurement, start, end, interval,filters, callback,client);
-  }
+
+/**
+* @desc Writes points into database
+*
+* @param data - Dictionary of params that were passed in url query
+*
+* @returns Promise - That the data was stored into the database.
+*/
+module.exports.storeData = function(data) {
+  // Get GitHub Data
+  var measurement = data[0].measurement;
+  var owner = data[0].requiredFilters.owner;
+  var repo = data[0].requiredFilters.repo;
+
+  return github.queryGithubData(measurement, owner, repo).then(points => {
+    console.log(points);
+    client.writePoints(points).catch(err => {
+      console.log('Error writing to db: ' + err);
+    });
+
+  })
+  .catch(err => {
+    console.log('There was an error using GitHub API.' + err);
+  });
 };
 
+/**
+* @desc Fetch data from database based on params given.
+*
+* @param data - Dictionary of params that were passed in url query.
+*
+* @returns Promise - The future data from the database.
+*/
+module.exports.fetchData = function(data) {
+  var requiredFilterQuery = '';
+  var optionalFilterQuery = '';
 
-function getGithubData(measurement, start, end, interval,filters, callback,client){
-  var filterQuery="";
+  for (var filter in data[0].requiredFilters){
+    if (filter !== 'owner'){
+      requiredFilterQuery += "AND " + filter + " = '" + data[0].requiredFilters[filter] + "' ";
+    }
+  }
 
-  for (var filter in filters) {
-    filterQuery+="AND " +filter+ " = '" +  filters[filter] +"' ";
+  for (var filter in data[0].optionalFilters){
+    if (filter !== 'owner'){
+      optionalFilterQuery += "OR " + filter + " = '" + data[0].optionalFilters[filter] + "' ";
+    }
   }
 
-  var query = 'SELECT sum(value) FROM ' + measurement
-            + ' WHERE time >= ' + start + '000000000 and time <= ' + end + '000000000 '
-            + filterQuery + 'group by time(' + interval + ') fill(0)';
+  var query = 'SELECT sum(value) FROM ' + data[0].measurement +
```

Comment:
In `time_series.js` I put all the given parameters into an array to have one argument to pass for the `query` functions instead of having multiple ones. I'll change it to an object so it's cleaner.